### DISCUSSION TITLE
How to control these macro definitions in the shader?

### DISCUSSION THREAD

**Discussion by dugufly** [google/filament#4642] [Q&A] [answered]
Where is this correspondence implemented for uberShader？ 

from #4621 

> You don't control those directly, they are inferred from what you do in your material. For instance if you specify a material.sheenColor, MATERIAL_HAS_SHEEN_COLOR` will be set. Those are implementation details you don't need to worry about. Everything public is documented in https://google.github.io/filament/Materials.html

**Comment by romainguy** [comment, marked as answer]
It's implemented in the material compiler (`libs/filamat`). These macros are not part of the public APIs.
